nin/dasBoot/NodeManager.js

```diff
@@ -5,22 +5,29 @@ class NodeManager {
   }
 
   createNode(nodeInfo) {
-    nodeInfo.options = nodeInfo.options || {};
+    const deepClonedNodeInfo = JSON.parse(JSON.stringify(nodeInfo));
+    deepClonedNodeInfo.options = deepClonedNodeInfo.options || {};
 
-    const nodeConstructor = nodeInfo.type.slice(0, 4) === 'NIN.' ?
-      NIN[nodeInfo.type.slice(4)]
-      : window[nodeInfo.type];
+    const nodeConstructor = deepClonedNodeInfo.type.slice(0, 4) === 'NIN.' ?
+      NIN[deepClonedNodeInfo.type.slice(4)]
+      : window[deepClonedNodeInfo.type];
 
     if (nodeConstructor === undefined) {
       return null;
     }
 
-    return new nodeConstructor(nodeInfo.id, nodeInfo.options);
+    return new nodeConstructor(deepClonedNodeInfo.id, deepClonedNodeInfo.options);
   }
 
   insertOrReplaceNode(node) {
     const nodeToReplace = this.nodes[node.id];
     if(nodeToReplace) {
+      for(let inputKey in nodeToReplace.inputs) {
```

Comment:
Did this just never happen before?

Reply:
It did before-before, but then a recent change fixed some other bug and introduced this regression.